### DISCUSSION TITLE
Dynamic Tags with hydration directives

### DISCUSSION THREAD

**Discussion by vjcspy** [withastro/roadmap#484] [Proposal]
### Body

# Summary

Dynamic tag is a very necessary feature for us to build dynamic layout feature. But unfortunately it can only be used with astro components.

# Background & Motivation

Our company needs to build a feature that can decide which components to use based on the user's configuration. Using dynamic tag is a prerequisite but unfortunately dynamic tag does not work with hydration directives.
I tried with some other frameworks and they seem to do this well (e.g. nextjs with dynamic import). But Astro is our priority with island architecture.

# Goals

Support dynamic tag when using the hydration directives

**Comment by matthewp** [comment]
This isn't possible due to the way that Astro does partial hydration. The way it knows how to load, for example, `<Client client:load />` is that it looks for `Client` as one of your imports, for example `import Client from '../components/Client.jsx'`. 

Dynamic references prevent Astro from doing that, so it would have no idea how to load the component.
